pylint/checkers/spelling.py

```diff
@@ -321,14 +323,15 @@ def open(self):
         )
         self.initialized = True
 
-    def close(self):
+    def close(self) -> None:
         if self.private_dict_file:
             self.private_dict_file.close()
 
-    def _check_spelling(self, msgid, line, line_num):
+    def _check_spelling(self, msgid: str, line: str, line_num: int) -> None:
         original_line = line
         try:
-            initial_space = re.search(r"^[^\S]\s*", line).regs[0][1]
+            # The mypy warning is caught by the except statement
+            initial_space = re.search(r"^\s+", line).regs[0][1]  # type: ignore[union-attr]
```

Comment:
There's a simplification in the regex here on top of the mypy ignore. ``[^\S]`` is ``\s`` and ``\s\s*`` is ``\s+``